js/chat/chat.ts

```diff
@@ -528,11 +528,21 @@ class ChatContainer extends LightElement {
 
 // ------- Register custom elements and shiny bindings ---------
 
-customElements.define(CHAT_MESSAGE_TAG, ChatMessage);
-customElements.define(CHAT_USER_MESSAGE_TAG, ChatUserMessage);
-customElements.define(CHAT_MESSAGES_TAG, ChatMessages);
-customElements.define(CHAT_INPUT_TAG, ChatInput);
-customElements.define(CHAT_CONTAINER_TAG, ChatContainer);
+if (!customElements.get(CHAT_MESSAGE_TAG)) {
+  customElements.define(CHAT_MESSAGE_TAG, ChatMessage);
+}
+if (!customElements.get(CHAT_USER_MESSAGE_TAG)) {
+  customElements.define(CHAT_USER_MESSAGE_TAG, ChatUserMessage);
+}
+if (!customElements.get(CHAT_MESSAGES_TAG)) {
+  customElements.define(CHAT_MESSAGES_TAG, ChatMessages);
+}
+if (!customElements.get(CHAT_INPUT_TAG)) {
+  customElements.define(CHAT_INPUT_TAG, ChatInput);
+}
+if (!customElements.get(CHAT_CONTAINER_TAG)) {
+  customElements.define(CHAT_CONTAINER_TAG, ChatContainer);
+}
```

Comment:
Now that we're here, maybe we take it one step further?

```suggestion
const chatCustomElements = [
  { tag: CHAT_MESSAGE_TAG, component: ChatMessage },
  { tag: CHAT_USER_MESSAGE_TAG, component: ChatUserMessage },
  { tag: CHAT_MESSAGES_TAG, component: ChatMessages },
  { tag: CHAT_INPUT_TAG, component: ChatInput },
  { tag: CHAT_CONTAINER_TAG, component: ChatContainer }
];

chatCustomElements.forEach(({ tag, component }) => {
  if (!customElements.get(tag)) {
    customElements.define(tag, component);
  }
});
```